src/components/MainNav/MyProfile/MyProfile.js

```diff
@@ -47,6 +55,38 @@ export default class MyProfile extends Component {
     this.getProfileImage = this.getProfileImage.bind(this);
   }
 
+  componentDidMount() {
+    const { modules, stripes } = this.props;
+    if (stripes.user) {
+      const userDropdownLinks = ([].concat(modules.app, modules.settings))
+        .filter(({ links }) => links && Array.isArray(links.userDropdown))
+        .reduce((result, { links }) => result.concat(links.userDropdown), []);
+
+      userDropdownLinks.forEach((link, index) => {
+        const linkFunction = link.check;
+        if (!linkFunction) {
+          this.createLink(link, index);
+        } else if (typeof userDropodownChecks[linkFunction] === 'function') {
+          userDropodownChecks[linkFunction]().then((canShow) => {
```

Comment:
I still don't understand why we've introduced the complexity of asynchronous functions here, with the promises API and the consequent re-rendering. Surely this is a simple boolean? Why can't we just say
```
  if (userDropodownChecks[linkFunction]()) {
    this.createLink(link, index);
  }
```
?

Reply:
One of the requirements of the folio-org/ui-myprofile#3 is to check that user has its local password. We check it by making an async call to the server. Otherwise it is not possible.

Reply:
It is not acceptable to make a call to the server every time we generate the user menu.

And I am 100% certain it's not necessary. Do an SSO login in one session and a direct login in another. Poke around inside the two sessions' Stripes objects. I bet you'll find a difference.

And if you don't we can _make_ one, just by having the SSO login procedure set `stripes.thidsIsAnSSOSession` to true.